### PR TITLE
Add biweekly option and tests

### PR DESCRIPTION
**Pull request by akhanaton**
This adds the biweekly option and makes sure that accounts cannot receive emails more than twice in any single month. Also added some tests to cover that.

### CODE REVIEW

File: lib/adoptoposs/jobs/policy.ex [hunk 1]
```diff
@@ -14,6 +14,18 @@ defmodule Adoptoposs.Jobs.Policy do
...
   end
 
+  @doc """
+  Authorizes to send biweekly emails if the given date is on the `permitted_weekday`.
+  """
+  def authorize(:send_emails_biweekly, %Date{} = date, permitted_weekday) do
+    day = date.day
+
+    case matches_weekday?(date, permitted_weekday) do
+      true when day <= 7 or day in 14..20 -> true
```
Comment (floriank) [inline lib/adoptoposs/jobs/policy.ex:24]: Just for style points, I'd recommend breaking the `or` statement into another case condition _or_ use an `if` clause, since using `case` on a boolean return seems a bit overkill :wink:

Comment (akhanaton) [reply lib/adoptoposs/jobs/policy.ex:24]: @floriank , thanks for the feedback. Will make the recommended changes. 👍

Comment (paulgoetze) [reply lib/adoptoposs/jobs/policy.ex]: Thanks a lot for your pull request @akhanaton!
I actually quite liked the one-liner in the case statement from before :wink: 

However, I found it kind of hard to grasp with the case statement. Thinking of our future selves, maybe it will be easier to read without it? Maybe even with day ranges (1..7 and 14..20) Something like:

```elixir
(day in @first_week || day in @third_week) && matches_weekday?(date, permitted_weekday)
```
And am I mistaken or would 14..20 be the range from the last day of the 2nd week to the last but one of the 3rd week? So since we are starting the month’s days with 1, not 0, should it be 15..21?

File: test/adoptoposs/jobs_test.exs [hunk 2]
```diff
@@ -52,6 +52,36 @@ defmodule Adoptoposs.JobsTest do
...
+
+    test "send_emails_biweekly is forbidden if the given date is not week 1 or week 3" do
+      # First authorized day within the first two weeks of the month
+      {:ok, date} = Date.new(2222, 01, 08)
+
+      assert {:error, :unauthorized} = Bodyguard.permit(Jobs, :send_emails_biweekly, date, 2)
+
+      assert {:error, :unauthorized} = Bodyguard.permit(Jobs, :send_emails_biweekly, date, "2")
+
+      # First authorized day within the first two weeks of the month
```
Comment (paulgoetze) [inline test/adoptoposs/jobs_test.exs:78]: Great tests! The only thing I’d have to remark is that I think the comments are a bit misleading.

Relating to the question on ranges for week 3 in my other comment –  could you add some test cases for the edge cases of the implementation to make sure we got the right day ranges?

That would be, it is forbidden for the: 
- permitted weekday, but if it is the first/last day of week 2
- permitted weekday, but if it is the first day of the week 4
- not permitted weekday in a permitted week

**Comment by paulgoetze** [comment]
@akhanaton Could you try [enabling GitHub actions](https://github.com/akhanaton/adoptoposs/actions) on your forked repo? I think else the tests won’t run automatically for this pull request.

File: lib/adoptoposs/jobs/policy.ex (hunk 1 above)
Comment (akhanaton) [reply lib/adoptoposs/jobs/policy.ex]: @paulgoetze , thanks for the feedback. I like your suggestion, it is a lot easier to figure out what's going on. You were correct on the date range (15..21).

File: test/adoptoposs/jobs_test.exs (hunk 2 above)
Comment (akhanaton) [reply test/adoptoposs/jobs_test.exs:78]: @paulgoetze ,  added a few more test and reworded some of the comments.

**Comment by akhanaton** [comment]
> @akhanaton Could you try [enabling GitHub actions](https://github.com/akhanaton/adoptoposs/actions) on your forked repo? I think else the tests won’t run automatically for this pull request.

done

File: lib/adoptoposs/jobs/policy.ex (hunk 1 above)
Comment (floriank) [reply lib/adoptoposs/jobs/policy.ex]: TIL 😉
